src/parser/xasl_generation.c

```diff
@@ -21924,9 +21881,14 @@ parser_generate_xasl (PARSER_CONTEXT * parser, PT_NODE * node)
 	  /* XASL cache related information */
 	  pt_init_xasl_supp_info ();
 
+	  db_query_plan_dump_fp_open ();
```

Comment:
pt_to_insert_xasl() also calls parser_generate_xasl_post().
It is necessary to check whether the plan of the query below is normal.
```
create table t (col1 int, col2 int);
set optimization level 513;
insert /*+ recompile */ into t values(1,(select 1 from dual limit 1));
```

Reply:
Query plan is shown as following
```
csql> insert /*+ recompile */ into t values(1,(select 1 from dual limit 1));
Join graph segments (f indicates final):
seg[0]: [0]
Join graph nodes:
node[0]: dual dual(0/1) (loc 0)
Join graph terms:
term[0]: inst_num() range (min inf_le 1) (sel 0.1) (rank 3) (instnum term) (not-join eligible) (loc 0)

Query plan:

sscan
    class: dual node[0]
    sargs: term[0]
    cost:  1 card 0

Query stmt:

(select 1 from dual dual where (inst_num()<= ?:0 ))
```

Additionally, I've tested too for the case that the stored procedure is called.
```
set optimization level 513;
create table t2 (col1 int, col2 string);
insert /*+ recompile */ into t2 values(1,(select hello() from dual limit 1)); -- hello() returns string type
```

Result
```
Join graph segments (f indicates final):
seg[0]: [0]
Join graph nodes:
node[0]: dual dual(0/1) (loc 0)
Join graph terms:
term[0]: inst_num() range (min inf_le 1) (sel 0.1) (rank 3) (instnum term) (not-join eligible) (loc 0)

Query plan:

sscan
    class: dual node[0]
    sargs: term[0]
    cost:  1 card 0

Query stmt:

(select hello() from dual dual where (inst_num()<=1))

```

Reply:
In case of insert query, db_query_plan_dump_fp_open() does not seem to be called differently than before. is this ok?

Reply:
I think yes. Insert query doesn't generate XASL, so opening a file pointer for the query plan is not necessary.